lib/mongo/collection.rb

```diff
@@ -341,7 +341,7 @@ def insert_one(document, options = {})
           :coll_name => name,
           :write_concern => write_concern,
           :bypass_document_validation => !!options[:bypass_document_validation],
-          :options => options
+          :id_generator => client.options[:id_generator]
```

Comment:
we are sure we don't need to pass :options to Operation::Write::Insert#initialize ?